api/v1alpha1/k8s_types.go

```diff
@@ -103,6 +103,19 @@ type PersistentVolumeClaimSpec struct {
 	VolumeMode *PersistentVolumeMode `json:"volumeMode,omitempty"`
 }
 
+// VolumeSpec contains the Persistent Volume specification for a benchmark run
+type VolumeSpec struct {
+	// VolumeSource represents the source of the volume, e.g. an existing
+	// PVC, host path, git repo, etc.
+	VolumeSource corev1.VolumeSource `json:"volumeSource,omitempty"`
```

Comment:
One more: VolumeSource is passed by value, while PersistentVolumeClaim is passed by reference. Why is it different?

Reply:
Passing PVC by reference enables us signal if it is unset.
If we'd pass by value we would need to match to an empty / defaulted PersistentVolumeClaim, which isn't as clean as checking for nil pointer.